### PR TITLE
Automation tests for Storybook components

### PR DESCRIPTION
**Pull request by DlgSHi**
**Added to testing scope:**

- InlineInputs,
- Link,
- MenuList,
- Menu

### CODE REVIEW

File: cypress/locators/menu-list/index.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import {
+  MENU_LIST_PREVIEW, MENU_TITLE, MENU_LIST_SEARCH, MENU_LIST,
+} from './locators';
+
+// component preview locators
+export const menuListPreview = () => cy.iFrame(MENU_LIST_PREVIEW);
+export const menuTitle = () => cy.iFrame(MENU_TITLE);
```
Comment (marcinstachowicz) [inline cypress/locators/menu-list/index.js]: `.first()` should be here, not in step def

File: cypress/locators/menu-list/index.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import {
+  MENU_LIST_PREVIEW, MENU_TITLE, MENU_LIST_SEARCH, MENU_LIST,
+} from './locators';
+
+// component preview locators
+export const menuListPreview = () => cy.iFrame(MENU_LIST_PREVIEW);
```
Comment (marcinstachowicz) [inline cypress/locators/menu-list/index.js]: `.children()` should be here, not in step def, `menuListPreview` can be renamed to `menuListChildren` or similar

File: cypress/support/step-definitions/menu-list-steps.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+Then('filter is enabled', () => {
+  menuListSearchInput().should('exist');
+});
+
+When('I change search parameter to {string}', (parameter) => {
+  menuListSearchInput().clear().type(parameter);
+});
+
+Then('search result is {string}', (parameter) => {
+  menuList().parent().should('have.length', 2);
```
Comment (marcinstachowicz) [inline cypress/support/step-definitions/menu-list-steps.js]: why `menuList().parent()` should have length equals 2? no information about it in step name or scenario

File: cypress/integration/link.feature [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+      | children                 |
+      | Sample text              |
+      | 1234567890               |
+      | áéíóú¿¡üñ                |
+      | !@#$%^*()_+-=~[];:.,?{}  |
+      | ÄÖÜßäöüß                 |
+      | <>                       |
+
+  @positive
+  Scenario: Enable Link component
```
[the remark this answers is not in the document]
Comment (DlgSHi) [reply cypress/integration/link.feature]: disable, fixed

File: cypress/integration/link.feature [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+    When I disable Link component
+      And I enable Link component
+    Then Link is enabled
+
+  @positive
+  Scenario Outline: Change Link href
+    When I set href to "<href>"
+    Then Link on preview href is set to "<href>"
+    Examples:
+      | href                 |
```
[the remark this answers is not in the document]
Comment (DlgSHi) [reply cypress/integration/link.feature]: fixed

File: cypress/integration/link.feature [hunk 6]
```diff
@@ -0,0 +1,212 @@
...
+      | Sample text              |
+      | 1234567890               |
+      | áéíóú¿¡üñ                |
+      | !@#$%^*()_+-=~[];:.,?{}  |
+      | ÄÖÜßäöüß                 |
+      | <>                       |
+
+  @positive
+  Scenario Outline: Change type of icon for a Link component
```
[the remark this answers is not in the document]
Comment (DlgSHi) [reply cypress/integration/link.feature]: agree with U, divided those tests

File: cypress/integration/menu.feature [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+  Scenario Outline: Change Menu 'as' property
+    When I select as to "<as>"
+    Then Menu as property on preview is "<as>"
+    Examples:
+      | as        |
+      | primary   |
+      | secondary |
+
+  @positive
+  Scenario: Check the persistance of Menu element
```
[the remark this answers is not in the document]
Comment (DlgSHi) [reply cypress/integration/menu.feature]: 1. fixed
2. added

File: cypress/integration/menuList.feature [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+      | áéíóú¿¡üñ                |
+      | !@#$%^*()_+-=~[];:.,?{}  |
+      | ÄÖÜßäöüß                 |
+      | <>                       |
+
+  @ignore
+  # @positive doesn't work on Carbon Demo site
+  Scenario Outline: Change MenuList filterPlaceholder
+    When I set filterPlaceholder to "<text>"
+    # Then filterPlaceholder on preview is "<text>"
```
[the remark this answers is not in the document]
Comment (DlgSHi) [reply cypress/integration/menuList.feature]: fixed

File: cypress/integration/menuList.feature [hunk 9]
```diff
@@ -0,0 +1,77 @@
+Feature: Menu list component
+  I want to change MenuList component properties
```
[the remark this answers is not in the document]
Comment (DlgSHi) [reply cypress/integration/menuList.feature]: fixed

File: cypress/integration/menuList.feature [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+
+  @ignore
+  # @positive doesn't work on Carbon Demo site
+  Scenario: Disable and enable innitially open
+    When I check initiallyOpen checkbox
+      And I uncheck initiallyOpen checkbox
+    Then MenuList component is not expanded
+
+  @positive
+  Scenario: Enable fiter
```
[the remark this answers is not in the document]
Comment (DlgSHi) [reply cypress/integration/menuList.feature]: fixed

File: cypress/locators/menu-list/index.js (hunk 1 above)
Comment (DlgSHi) [reply cypress/locators/menu-list/index.js]: fixed

File: cypress/locators/menu-list/index.js (hunk 2 above)
Comment (DlgSHi) [reply cypress/locators/menu-list/index.js]: fixed

File: cypress/support/step-definitions/menu-list-steps.js (hunk 3 above)
Comment (DlgSHi) [reply cypress/support/step-definitions/menu-list-steps.js]: it is just additional assertion, which checks that there are the search input and the found element

File: cypress/integration/menuList.feature [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+      | text                     |
+      | Sample text              |
+      | 1234567890               |
+      | áéíóú¿¡üñ                |
+      | !@#$%^*()_+-=~[];:.,?{}  |
+      | ÄÖÜßäöüß                 |
+      | <>                       |
+
+  @positive
+  Scenario Outline: Check search field
```
Comment (marcinstachowicz) [inline cypress/integration/menuList.feature]: Here is what I suggest:

```suggestion
 Scenario Outline: Check search field
    When I click into menu item second element
      And I change search parameter to "<parameter>"
    Then search result is "<result>"
      And resuts count is <resultsCount>
    Examples:
      | parameter | result                                        | resultsCount |
      | Fir       | First Sub Item                                | 1            |
      | Sec       | Second Sub Item                               | 1            |
      | Thi       | Third Sub Item                                | 1            |
      | d         | Second Sub Item Third Sub Item                | 2            |
      | tem       | First Sub Item Second Sub Item Third Sub Item | 3            |
```

File: cypress/support/step-definitions/menu-list-steps.js (hunk 3 above)
Comment (marcinstachowicz) [reply cypress/support/step-definitions/menu-list-steps.js]: I suggest to add separate step for results count assertion which I put it as scenario comment, please see there and tell me what you think :)

Comment (DlgSHi) [reply cypress/support/step-definitions/menu-list-steps.js]: added

File: cypress/integration/menuList.feature (hunk 11 above)
Comment (DlgSHi) [reply cypress/integration/menuList.feature]: done

**Pull request merged**